addons/mail/controllers/discuss.py

```diff
@@ -541,8 +541,14 @@ def audio_worklet_processor(self):
     # --------------------------------------------------------------------------
 
     @http.route('/mail/guest/update_name', methods=['POST'], type='json', auth='public')
-    def mail_guest_update_name(self, name):
-        guest = request.env['mail.guest']._get_guest_from_request(request)
-        if not guest:
-            raise NotFound()
-        guest.sudo()._update_name(name)
+    def mail_guest_update_name(self, guest_id, name):
+        if request.env.user._is_admin():
+            guest_to_rename = request.env['mail.guest'].browse(guest_id).exists()
+            if not guest_to_rename:
+                raise NotFound()
+            guest_to_rename._update_name(name)
+        else:
+            guest = request.env['mail.guest']._get_guest_from_request(request)
+            if not guest or guest_id != guest.id:
+                raise NotFound()
+            guest.sudo()._update_name(name)
```

Comment:
This looks like it should be 2 separate method with a such if/else case.

Or actually use the same code in both case, with just a minor variation for the access check.

`guest_to_rename = request.env['mail.guest'].browse(guest_id).exists()` and `guest.sudo()._update_name(name)` should be common top level code.

Reply:
adjusted according to your second suggestion :ok_hand: